presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreValueUtil.java

```diff
@@ -0,0 +1,221 @@
+            }
+            else if (type instanceof VarbinaryType) {
+                byte[] x = null;
+                if (value instanceof byte[]) {
+                    x = (byte[]) value;
+                }
+                else if (value instanceof String) {
+                    x = ((String) value).getBytes(UTF_8);
+                }
+                else if (value instanceof Slice) {
+                    x = ((Slice) value).getBytes();
+                }
+                if (x != null) {
+                    if (toPrimaryKeyNotColumnValue) {
+                        return (T) PrimaryKeyValue.fromBinary(x);
+                    }
+                    else {
+                        return (T) ColumnValue.fromBinary(x);
+                    }
+                }
+                acceptedTypes = new Class[] {String.class, byte[].class, Slice.class};
+            }
+        }
+        catch (Throwable e) {
+            String x = String.valueOf(value);
+            String str = "convert() failed for " + (toPrimaryKeyNotColumnValue ? "PrimaryKeyValue" : "ColumnValue") +
+                    ", required type is " + type + ", and type of value is " + value.getClass() + ", and value=";
+            log.error(str + x, e); //全部打印出来
+
+            String y = x;
+            if (x.length() > 20) {
+                y = x.substring(0, 20) + "......";
+            }
+            throw new IllegalArgumentException(str + y, e); //打印部分
+        }
+
+        String str = "Unsupported conversion for " + (toPrimaryKeyNotColumnValue ? "PrimaryKeyValue" : "ColumnValue") + ", ";
+
+        if (acceptedTypes == null) {
+            str += "the required tablestore type isn't supported not:" + type;
+        }
+        else {
+            str += "the required tablestore type is " + type + ", but ";
+            if (value == null) {
+                str += "the value is NULL";
+            }
+            else {
+                str += "the type of value is " + value.getClass();
+            }
+            str += " and accepted java types is " + Arrays.toString(acceptedTypes);
+        }
+        throw new UnsupportedOperationException(str);
+    }
+
+    public static PrimaryKeyValue toPrimaryKeyValue(Marker marker)
+    {
+        return convert(marker.getType(), true, marker.getValue());
+    }
+
+    public static PrimaryKeyValue addOneForPrimaryKeyValue(PrimaryKeyValue target)
+    {
+        if (target == INF_MAX || target == INF_MIN) {
+            return target;
+        }
+        switch (target.getType()) {
+            case INTEGER:
+                long v1 = target.asLong();
+                if (v1 == Long.MAX_VALUE) {
+                    return INF_MAX;
+                }
+                return fromLong(v1 + 1);
+            case STRING:
+                String v2 = target.asString();
+                return PrimaryKeyValue.fromString(v2 + "\0");
+            case BINARY:
+                byte[] v3 = target.asBinary();
```

Comment:
I don't think v1, v2 and v3 are very meaningful name here.

Reply:
I agree with you, changed it.